src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNVariantTrain.java

```diff
@@ -0,0 +1,119 @@
+package org.broadinstitute.hellbender.tools.walkers.vqsr;
+
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.programgroups.ShortVariantDiscoveryProgramGroup;
+import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.python.PythonScriptExecutor;
+import picard.cmdline.programgroups.VariantEvaluationProgramGroup;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Train a Convolutional Neural Network (CNN) for filtering variants.
+ * This tool expects training data generated by {@link CNNVariantWriteTensors}.
+ *
+ *
+ * <h3>Inputs</h3>
+ * <ul>
+ *      <li>data-dir The training data created by {@link CNNVariantWriteTensors}.</li>
+ *      <li>The tensor-name argument determines what types of tensors the model will expect.
+ *      Set it to "reference" for 1D tensors or "read_tensor" for 2D tensors.</li>
+ * </ul>
+ *
+ * <h3>Outputs</h3>
+ * <ul>
+ * <li>output-dir The model weights file and semantic configuration json are saved here.
+ *  This default to the current working directory.</li>
+ * <li>model-name The name for your model.</li>
+ * </ul>
+ *
+ * <h3>Usage example</h3>
+ *
+ * <h4>Train a 1D CNN on Reference Tensors</h4>
+ * <pre>
+ * gatk CNNVariantTrain \
+ *   -tensor-name reference \
+ *   -data-dir my_tensor_folder \
+ *   -model-name my_1d_model
+ * </pre>
+ *
+ * <h4>Train a 2D CNN on Read Tensors</h4>
+ * <pre>
+ * gatk CNNVariantTrain \
+ *   -data-dir my_tensor_folder \
+ *   -tensor-name read_tensor \
+ *   -model-name my_2d_model
+ * </pre>
+ *
+ */
+@CommandLineProgramProperties(
+        summary = "Train a CNN model for filtering variants",
+        oneLineSummary = "Train a CNN model for filtering variants",
+        programGroup = VariantEvaluationProgramGroup.class
+)
+@DocumentedFeature
+@ExperimentalFeature
+public class CNNVariantTrain extends CommandLineProgram {
+
+    @Argument(fullName = "data-dir", shortName = "dd", doc = "Directory of training tensors, created if write-tensors is true, otherwise read.", optional = true)
+    private String dataDir = "";
+
+    @Argument(fullName = "tensor-name", shortName = "tn", doc = "Name of the tensors to generate, reference for 1D reference tensors and read_tensor for 2D tensors.", optional = true)
+    private String tensorName = "reference";
```

Comment:
We should use an enum here, and in CNNWriteTensors, then, instead of a String for this argument. In addition to being validated by the parser and generally being type-safe, the command line and help systems will display the possible values for the user. If you make it a `ClpEnum` (we don't have any of those on GTAK, but you can see one [here]( https://github.com/broadinstitute/picard/blob/e0e659f039a051d29b58680337601d754b6eee55/src/main/java/picard/sam/SortSam.java#L119), each value in the enum can have a doc string describing it in more detail.

Reply:
ooo I like this, will do